parser_tool/static/js/src/case_distribution.js

```diff
@@ -0,0 +1,302 @@
+                                .style("fill-opacity", .7);
+                        })
+                        .on('mouseout', function () {
+                            tooltip
+                                .transition(200)
+                                .style('opacity', 0);
+                            g.selectAll("polygon")
+                                .transition(200)
+                                .style("fill-opacity", cfg.opacityArea);
+                        })
+                        .append("svg:title")
+                        .text(function (j) { return Math.max(j.value, 0) });
+
+                    series++;
+                });
+                //Tooltip
+                tooltip = g.append('text')
+                    .style('opacity', 0)
+                    .style('font-family', 'sans-serif')
+                    .style('font-size', '20px');
+            }
+        };
+        return RadarChart
+    }
+    async function make_chart(data, lemma_lookup) {
+        $("#error").hide();
+        var input_text = $("#contentinput").val().trim();
+        var index = lemma_lookup[input_text];
+
+        if (index == undefined) {
+            $("#output").hide();
+            $("#error").show();
+            return false;
+        }
+
+        var radar_data = data[index];
+        $("#lemma").text(input_text);
+        $("#output").show();
+
+        get_radar_config().then(function (result) {
+            var RadarChart = result;
+
+            var w = 500,
+                h = 500;
+
+            var colorscale = d3.scaleOrdinal(d3.schemeCategory10);
+
+            //Legend titles
+            var LegendOptions = ['Singular', 'Plural'];
+
+            //Data
+            var d = [
+                radar_data.data[0].axes, radar_data.data[1].axes
+            ]
+
+            // get count totals
+            var singular_total = 0;
+            var plural_total = 0;
+
+            for (var val in d[0]) {
+                singular_total += d[0][val]["value"];
+            }
+            for (var val in d[1]) {
+                plural_total += d[1][val]["value"];
+            }
+            $("#sg").text(singular_total);
+            $("#pl").text(plural_total);
+            
+            //Options for the Radar chart, other than default
+            var mycfg = {
+                w: w,
+                h: h,
+                maxValue: 0.6,
+                levels: 6,
+                ExtraWidthX: 300
+            }
+
+            //Call function to draw the Radar chart
+            //Will expect that data is in %'s
+            RadarChart.draw("#chart", d, mycfg);
+        })
+    };
+
+    $(document).ready(async function () {
+        console.log("ready!");
+        $("#output").hide();
+        $.getJSON("/static/js/src/data/RNC.json", function (data) {
+            // create lookup from lemma to array number to avoid looping every time
+            var lemma_lookup = new Object();
```

Comment:
Note that this could also be done like this since you're just plucking the "lemma" from each item in the array and indexing its position:

```javascript
const lemma_lookup = data.reduce(function(acc, cur, idx, arr) {
    acc[cur.lemma] = idx;  
    return acc;
}, {});
```

I'm not sure if this will perform better/worse than the straight for loop given the amount of data. But this reads a bit clearer, I think.